samples/M5StackRemoteDisplay/RemoteControl.cs

```diff
@@ -0,0 +1,554 @@
+                {
+                    return sentence.Speed.ToUnit(SpeedUnit.Knot);
+                }
+
+                return null;
+            }));
+
+            _dataSets.Add(new NmeaValueDataSet("Course over Ground", s =>
+            {
+                if (_positionProvider.TryGetCurrentPosition(out _, out Angle track, out _, out _))
+                {
+                    return track;
+                }
+
+                return null;
+            }, "F1"));
+
+            _dataSets.Add(new NmeaValueDataSet("Heading", s =>
+            {
+                if (s.TryGetLastSentence(HeadingTrue.Id, out HeadingTrue sentence))
+                {
+                    return sentence.Angle;
+                }
+
+                return null;
+            }, "F1"));
+
+            _dataSets.Add(new NmeaValueDataSet("Depth below Surface", s =>
+            {
+                if (s.TryGetLastSentence(DepthBelowSurface.Id, out DepthBelowSurface sentence))
+                {
+                    return sentence.Depth.ToUnit(LengthUnit.Meter);
+                }
+
+                return null;
+            }));
+
+            _dataSets.Add(new NmeaValueDataSet("Engine RPM", s =>
+            {
+                if (s.TryGetLastDinSentence(SeaSmartEngineFast.HexId, out SeaSmartEngineFast sentence))
+                {
+                    return sentence.RotationalSpeed.ToUnit(RotationalSpeedUnit.RevolutionPerMinute);
+                }
+
+                return null;
+            }));
+
+            _dataSets.Add(new NmeaValueDataSet("Pitch", s =>
+            {
+                if (s.TryGetTransducerData("PITCH", out TransducerDataSet set))
+                {
+                    return Angle.FromDegrees(set.Value);
+                }
+
+                return null;
+            }));
+
+            _dataSets.Add(new NmeaValueDataSet("Roll", s =>
+            {
+                if (s.TryGetTransducerData("ROLL", out TransducerDataSet set))
+                {
+                    return Angle.FromDegrees(set.Value);
+                }
+
+                return null;
+            }));
+
+            _selectedDataSet = 0;
+            _leftMouseMenuBar = BitmapImage.CreateFromFile("images/MenuBarLeftMouse.png");
+            _rightMouseMenuBar = BitmapImage.CreateFromFile("images/MenuBarRightMouse.png");
+            _defaultMenuBar = BitmapImage.CreateFromFile("images/MenuBar.png");
+            _openMenu = BitmapImage.CreateFromFile("images/OpenMenu.png");
+
+            _tcpClient.StartDecode();
+        }
+
+        private void OnNewSequence(NmeaSinkAndSource nmeaSinkAndSource, NmeaSentence nmeaSentence)
+        {
+            // Nothing to do here, handled by the message cache (doing this floods the output)
+            // Console.WriteLine($"Received sentence: {nmeaSentence.ToString()}");
+        }
+
+        private void OnTouched(object o, Point point)
+        {
+            Console.WriteLine($"Touched screen at {point}");
```

Comment:
are the Console.WriteLine really needed or could be replaced by Debug.WriteLine? Or just the logger?

Reply:
It's just debug output, yes, but the entire console is showing nothing but debug output anyway.

Reply:
yes and no :-) with Debug.WriteLine, you do not see those when you build in Release! So it's from my point of view, much cleaner than using the Console.